iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -3613,6 +3613,8 @@ void IoTHubTransport_MQTT_Common_DoWork(TRANSPORT_LL_HANDLE handle)
             }
             else if (transport_data->currPacketState == PUBLISH_TYPE)
             {
+                // The duplicated call here and down below is intentional at this point.
+                // This is the simplest way to guarantee compliance with MQTT v3.1.1 [MQTT-4.6.0-1].
```

Comment:
```suggestion
                // This is the simplest way to guarantee compliance with MQTT v3.1.1 [MQTT-4.6.0-1].
```
                // This is the simplest way to guarantee compliance with MQTT v3.1.1 [MQTT-4.6.0-1].
                // ** QoS1 Publish messages must be send in packet id order on reconnect